### ISSUE TITLE
New type of morph

### ISSUE THREAD

**Issue by SerendipityNL** [laravel/ideas#1755]
### Explanation
I'm trying to build a new type of morph (and combining all kinds of methods from other relationships has given me some working prototype, but it isn't a nice solution at all). It shifts the responsibility of the morph from the child to the parent. If you look at the documentation the child currently stores what it's bound to, while I'm trying to change that so the parent stores the class it's children will be.

My goal in the end is to be able to run this line:

    $entities = Entity::with('attributes.values')->find($id)

Considering the following EAV database setup:

    entity
        id

    attributes
        id
        type

    value_datetimes
        id
        attribute_id
        entity_id

    value_decimals
        id
        attribute_id
        entity_id

    value_integers
        id
        attribute_id
        entity_id

    value_varchars
        id
        attribute_id
        entity_id

What I'm trying to accomplish is saving the type of the attribute in the `attributes` table, so a value could be `App\ValueVarchars`. The current method of morphing does not allow the class to be stored in the same table items are being morphed to. 

The [example](https://laravel.com/docs/master/eloquent-relationships#one-to-many-polymorphic-relations) given in the documentation also has the `commentable` key, in which I would like to store the relationship.

So take the example from above, I would like to set the child classes in one function (or under one keyword) and be able to always select them in the same manner (always `$attribute->values`).

### Question
Am I approaching this totally wrong or is this indeed some logic Laravel could use?

**Comment by antonkomarev** [comment]
This looks very similar to Magento EAV.

**Comment by SerendipityNL** [comment]
> This DB structure looks very similar to Magento EAV.

This is true. I would like to use the structure, but without the big resource consuming system that Magento is these days.

**Comment by prezire** [comment]
I might have came across something similar. For example, I was looking into the constructor's 3rd parameter https://laravel.com/api/5.3/Illuminate/Database/Eloquent/Relations/MorphOne.html#method___construct hoping it does what's expected here. Although I haven't gone deep enough into Larvel's API codes yet.

**Comment by SerendipityNL** [comment]
@prezire Sorry for my late reaction, but the `morphOne` is not what I need. The problem with `morphOne` is that only a single value could be related to an attribute. I want multiple values to be related. The thing that is correct though is the class that's stored on the attribute of the to be related resource!
